src/sas/sascalc/fit/BumpsFitting.py

```diff
@@ -282,32 +284,56 @@ def fit(self, msg_q=None,
 
         # TODO: shouldn't reference internal parameters of fit problem
         varying = problem._parameters
+        # Propagate uncertainty through the parameter expressions
+        # We are going to abuse bumps a little here and stuff uncertainty
+        # objects into the parameter values, then update all the
+        # derived parameters with uncertainty propagation. We need to
+        # avoid triggering a model recalc since the uncertainty objects
+        # will not be working with sasmodels.
+        # TODO: if dream then use forward MC to evaluate uncertainty
+        # TODO: move uncertainty propagation into bumps
+        # TODO: should scale stderr by sqrt(chisq/DOF) if dy is unknown
+        values, errs, state = result['value'], result["stderr"], result[
+            'uncertainty']
```

Comment:
I still have to separate the case where only one parameter is being fitted since `np.cov` returns a 0 dimension array when a (1,n) array is passed as argument. This will cause `correlated_values` to fail since it is excepting at least a 1-d array.